sdk/resourcemanager/azure-resourcemanager-network/src/main/java/com/azure/resourcemanager/network/models/LoadBalancerOutboundRule.java

```diff
@@ -0,0 +1,226 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.network.models;
+
+import com.azure.core.annotation.Fluent;
+import com.azure.resourcemanager.network.fluent.models.OutboundRuleInner;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.ChildResource;
+import com.azure.resourcemanager.resources.fluentcore.model.Attachable;
+import com.azure.resourcemanager.resources.fluentcore.model.HasInnerModel;
+import com.azure.resourcemanager.resources.fluentcore.model.Settable;
+
+import java.util.List;
+
+/** An immutable client-side representation of an outbound rule. */
+@Fluent()
+public interface LoadBalancerOutboundRule
+    extends HasInnerModel<OutboundRuleInner>,
+    HasProtocol<LoadBalancerOutboundRuleProtocol>,
+    ChildResource<LoadBalancer> {
+
+
+    /** @return the associated frontend IP configuration ids */
+    List<String> frontendIpConfigurationIds();
+
+    /** @return the associated frontends */
+    List<LoadBalancerFrontend> frontends();
+
+    /** @return the associated backend address pool id */
+    String backendAddressPoolId();
+
+    /** @return the associated backend */
+    LoadBalancerBackend backend();
+
+    /** @return the number of outbound ports to be used for NAT */
+    int allocatedOutboundPortsNumber(); // TODO: need this or not?
```

Comment:
Will remove this Todo after making sure outbound ports will be allocated automatically